src/pages/layout.jsx

```diff
@@ -15,49 +15,70 @@ import {
   fetchExposures,
   fetchAlmanac,
   fetchNarrativeLog,
+  getDayobsStr,
+  getDatetimFromDayobsStr,
 } from "@/utils/fetchUtils";
 
 export default function Layout({ children }) {
-  const [exposures, setExposures] = useState(0);
-  const [dayObsStart, setDayObsStart] = useState(new Date(2025, 4, 12));
-  const [dayObsEnd, setDayObsEnd] = useState(new Date(2025, 4, 13));
+  const [exposureCount, setExposureCount] = useState(0);
+  const [dayobs, setDayobs] = useState(DateTime.utc().toJSDate());
+  const [noOfNights, setNoOfNights] = useState(1);
   const [instrument, setInstrument] = useState("LSSTCam");
 
   const [nightHours, setNightHours] = useState(0.0);
   const [weatherLoss, setWeatherLoss] = useState(0.0);
   const [sumExpTime, setSumExpTime] = useState(0.0);
   const [faultLoss, setFaultLoss] = useState(0.0);
 
-  const handleStartDateChange = (date) => {
-    setDayObsStart(date);
+  const handleDayobsChange = (date) => {
+    setDayobs(date);
+  };
+
+  const handleNoOfNightsChange = (nightsCount) => {
+    setNoOfNights(nightsCount.target.value);
```

Comment:
This nomenclature might be confusing, I recommend to change the parameter name to `event` instead:
```suggestion
  const handleNoOfNightsChange = (event) => {
    setNoOfNights(event.target.value);
```
Or go to the place where `handleNoOfNightsChange` is called and pass the `nightsCount` directly instead of passing the click event (i.e. unpack the target.value outside this function).